### ISSUE TITLE
some problem with yaml editor

### ISSUE THREAD

**Issue by wangxi761** [spring-projects/sts4#126]
i got a  jar in my project
<dependency>
    <groupId>org.springframework.cloud</groupId>
    <artifactId>spring-cloud-starter-bus-kafka</artifactId>
    <version>1.3.4.RELEASE</version>
</dependency>
and it should has a config **spring.cloud.stream.kafka.binder.***
but in generic ediotr, there is no prompt with it. some other jar has the same problem.in old version ,it is ok

another problem  in new version , i can't click ctrl+leftmouse  goto  the property class in yaml editor

**Comment by martinlippert** [comment]
The hyperlink navigation from the property editor to the class is indeed something that is missing from the new editor and on the list of things that we need to bring back. So stay tuned for future updates.

But I don't quite understand the first issue that you describe. I am confused by your mention of the generic editor, since the property files should be opened using the specific Spring Property editors in STS4 (which are based on the generic editor). Can you maybe provide a sample project that reproduces this problem? That would be awesome.

**Comment by wangxi761** [comment]
![image](https://user-images.githubusercontent.com/20579340/47552109-82293480-d936-11e8-81f1-5877031c15c4.png)
in this picture , we can see  a  property spring.cloud.stream.kafka.binder  ,it is declared in jar files. but  in yaml editor ,
![image](https://user-images.githubusercontent.com/20579340/47552284-e9df7f80-d936-11e8-90a2-ceaf2ee37683.png)
i can't find it @martinlippert

**Comment by wangxi761** [comment]
add another point ，if i  Ignore error tips，and use it forcibly， the project can run correctly, so think the editor may not detected this property

**Comment by martinlippert** [comment]
I tried to reproduce that by creating a new boot project using the initializr wizard and selected the Spring Cloud and the Spring Kafka Streams modules. The properties show up in the auto completion.

So I guess your project is somehow different from the one that I created. Can you attach a sample project that reproduces this issue?

**Comment by wangxi761** [comment]
i think it dose not need to attach a sample project  , if  i use this version
>  
    <dependency>
        <groupId>org.springframework.cloud</groupId>
        <artifactId>spring-cloud-starter-bus-kafka</artifactId>
        <version>2.0.0.RELEASE</version>
    </dependency>
i can find the property 
![image](https://user-images.githubusercontent.com/20579340/47585566-3a36fb80-d990-11e8-95ee-7d332a674f43.png)
but if i use the version 1.3.4.RELEASE and lower , such as the picture in  comment above

so ,in source code , you can find the property, but auto completion can not.

**Comment by kdvolder** [comment]
> but if i use the version 1.3.4.RELEASE and lower

Sounds to me like the older version of that dependency simply don't include the properties metadata. If that is the case then this is not a bug but simply the expected behavior. STS uses the metadata to give you content assist. So no metadata means no content assist.

However, I'm just guessing at the moment, I will see if I can confirm whether the guess is right.

**Comment by kdvolder** [comment]
I wasn't really able to confirm my guess. I'm not sure exactly what is in your pom. I tried changing some versions in the pom generated by 'New Spring Starter Project' but not really able to puzzle it out. Simply changing the version of this:

```
<dependency>
        <groupId>org.springframework.cloud</groupId>
        <artifactId>spring-cloud-stream-binder-kafka-core</artifactId>
</dependency>
```

To override the managed version with 1.3.4.RELEASE doesn't work. There is no 1.3.4.RELEASE version of this.

Also the project that I have from the wizard doesn't have `spring-cloud-starter-bus-kafka` in its dependency graph. So I can't try 'downgrading' it version.

Anyhow... I am going to close this ticket because as far as I know, this is simply because whatever stuff you are having in your pom exactly doesn't have metadata for the kafka binder properties on it. 

If this would be happening with a recent version the dependencies then I'd say we should file a bit with that dependency to fix their metadata. But since it seems to work fine with recent version, if there was a problem with in the old versions, it seems to already have been fixed.

In other words, I think everything is working as designed and there is nothing here that we can fix.

If you still think there is actually bug in STS, then please attach a *complete* project to reproduce the problem and we can take another look.
